File: kt/Minsk/src/main/kotlin/dev/phytolizer/minsk/UnaryExpressionSyntax.kt
package dev.phytolizer.minsk

class UnaryExpressionSyntax(val operatorToken: SyntaxToken, val operand: ExpressionSyntax) : ExpressionSyntax() {
    override val kind: SyntaxKind
        get() = SyntaxKind.UnaryExpression
    override val children: List<SyntaxNode>
        get() = listOf(operatorToken, operand)
}

File: nim/src/minsk/codeAnalysis/expressions/unaryExpressionSyntax.nim
import ../expressionSyntax
import ../syntaxKind
import ../syntaxNode
import ../syntaxToken

type
  UnaryExpressionSyntax* = ref object of ExpressionSyntax
    operatorToken*: SyntaxToken
    operand*: ExpressionSyntax

proc newUnaryExpressionSyntax*(
  operatorToken: SyntaxToken,
  operand: ExpressionSyntax
): UnaryExpressionSyntax =
  new(result)
  result.operatorToken = operatorToken
  result.operand = operand

method kind*(self: UnaryExpressionSyntax): SyntaxKind =
  return SyntaxKind.UnaryExpression

method children*(self: UnaryExpressionSyntax): seq[SyntaxNode] =
  result.add(self.operatorToken)
  result.add(self.operand)
